velox/functions/prestosql/Comparisons.cpp

```diff
@@ -94,6 +94,29 @@ struct SimdComparator {
     }
   }
 
+  template <typename T>
+  inline bool compare(T& l, T& r) const {
+    if constexpr (std::is_floating_point_v<T>) {
+      bool filtered = false;
+      if constexpr (std::is_same_v<ComparisonOp, std::equal_to<>>) {
```

Comment:
Instead of hard-coding std:equal_to<>, could we give it a name that can be both used here and passed to ComparisonSimdFunction<>? E.g.,
```
using Eq = std:equal_to<>;
...
VELOX_DECLARE_VECTOR_FUNCTION(
    udf_simd_comparison_eq,
    (ComparisonSimdFunction<Eq>::signatures()),
    (std::make_unique<ComparisonSimdFunction<Eq>>()));
...
template <typename T>
  inline bool compare(T& l, T& r) const {
    if constexpr (std::is_floating_point_v<T>) {
      bool filtered = false;
      if constexpr (std::is_same_v<ComparisonOp, Eq>) {
...
}}
```